python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -0,0 +1,1150 @@
+        Parameters
+    ----------
+    shape : dict of str to tuple, optional
+        The input shape to the graph
+    dtype : str or dict of str to str
+        The input types to the graph
+    """
+
+    current = None
+
+    def __init__(self, shape, dtype, target):
+        self._nodes = {}
+        self._inputs = {}
+        self._num_input = 0
+        self._shape = shape.copy() if shape else {}
+        self._input_names = []
+        self._dtype = dtype
+        self.opset = None
+        self._target = target
+        self.bb = relax.BlockBuilder()
+
+    def from_onnx(self, graph, opset) -> IRModule:
+        """Construct Relax expression from ONNX graph.
+        Onnx graph is a python protobuf object.
+        The companion parameters will be handled automatically.
+        However, the input names from onnx graph is vague, mixing inputs and
+        network weights/bias such as "1", "2"...
+        For convenience, we rename the `real` input names to "input_0",
+        "input_1"... And renaming parameters to "param_0", "param_1"...
+        Parameters
+        ----------
+        graph : onnx protobuf object
+            The loaded onnx graph
+        opset : opset version
+        Returns
+        -------
+        mod : tvm.IRModule
+            The returned relax module
+        params : dict
+            A dict of name: tvm.nd.array pairs, used as pretrained weights
+        """
+        with self.bb.function("main"):
+            with self.bb.dataflow() as df:
+                self.opset = opset
+                self._parse_graph_initializers(graph)
+                self._parse_graph_input(graph)
+                self._check_for_unsupported_ops(graph)
+                self._construct_nodes(graph)
+
+                # now return the outputs
+                outputs = [self._nodes[self._parse_value_proto(i)] for i in graph.output]
+                outputs = outputs[0] if len(outputs) == 1 else relax.Tuple(outputs)
+
+                ## Maintain the order of inputs and parameters from the ONNX graph, but only include
+                ## those parameters that are needed to execute the relax graph
+                nodes = {v: k for k, v in self._nodes.items()}
+                # Create a function from our output expression and all input variables.
+                param_list = [v for k, v in self._inputs.items()]
+                output_var = self.bb.emit_output(outputs)
+            self.bb.emit_func_output(output_var, params=param_list)
+        return self.bb.get()
+
+    def _parse_graph_initializers(self, graph):
+        """Parse network inputs to relax, aka parameters."""
+        for init_tensor in graph.initializer:
+            if not init_tensor.name.strip():
+                raise ValueError("Tensor's name is required.")
+            array = self._parse_array(init_tensor)
+            self._nodes[init_tensor.name] = relax.const(array)
+
+    def _parse_graph_input(self, graph):
```

Comment:
docstrings

Reply:
My personal opinion that the name of the function is pretty indicative of what it does. The implementation is also pretty straight forward. Given it is a private function and implementation detail, I am not sure if a docstring for this function will add much value here other than repeating what is obvious. Do you have a concrete suggestion?